File: app/src/main/java/com/agrogames/islandsofwar/engine/impl/unit/UnitAdder.java
package com.agrogames.islandsofwar.engine.impl.unit;

import com.agrogames.islandsofwar.engine.abs.unit.IUnit;
import com.agrogames.islandsofwar.engine.abs.unit.IUnitAdder;
import com.agrogames.islandsofwar.sounds.abs.SoundPlayer;

import java.util.ArrayList;
import java.util.List;

public class UnitAdder implements IUnitAdder {
    private final List<IUnit> units = new ArrayList<>();
    private final SoundPlayer soundPlayer;

    public UnitAdder(SoundPlayer soundPlayer) {
        this.soundPlayer = soundPlayer;
    }

    @Override
    public void addUnit(IUnit unit) {
        units.add(unit);
        soundPlayer.playSound("landing");
    }

    public List<IUnit> getUnits(){
        return units;
    }
}

File: app/src/main/java/com/agrogames/islandsofwar/engine/impl/unit/UnitAdder.kt
package com.agrogames.islandsofwar.engine.impl.unit

import com.agrogames.islandsofwar.engine.abs.unit.IUnit
import com.agrogames.islandsofwar.engine.abs.unit.IUnitAdder
import com.agrogames.islandsofwar.sounds.SoundPlayer

class UnitAdder() : IUnitAdder {
    override val units: MutableList<IUnit> = ArrayList()
    override fun addUnit(unit: IUnit) {
        units.add(unit)
        SoundPlayer.playSound("landing")
    }
}